document-portal/document-portal.c

```diff
@@ -1416,6 +1416,109 @@ portal_list (GDBusMethodInvocation *invocation,
   return TRUE;
 }
 
+const char *
+get_host_path_internal (GDBusMethodInvocation *invocation,
+                        XdpAppInfo            *app_info,
+                        const char            *id,
+                        GError                **error)
+{
+  g_autoptr(PermissionDbEntry) entry = NULL;
+  g_autoptr(GVariant) data = NULL;
+  const char *path = NULL;
+
+  XDP_AUTOLOCK (db);
+
+  entry = permission_db_lookup (db, id);
+
+  if (!entry)
+    {
+      if (error != NULL && *error == NULL)
+        {
+          g_set_error (error,
+                      XDG_DESKTOP_PORTAL_ERROR,
+                      XDG_DESKTOP_PORTAL_ERROR_INVALID_ARGUMENT,
+                      "%s", g_strdup_printf ("Invalid ID passed (%s)", id));
```

Comment:
```suggestion
                      "Invalid ID passed (%s)", id);
```
Shouldn't this work?